### PR TITLE
Configure fixes

### PR DESCRIPTION
**Pull request by karasevb**
Fixed problem in `make distclean` after configure with external `pmix, hwloc, libevent`:
```
./configure \
  --with-hwloc=/sandbox/hwloc \
  --with-libevent=/home/user/pmix/libevent-2.0.22-stable/install \
  --with-pmix=/sandbox/pmix \
  --prefix=`pwd`/install
```
Error log:
```
rm -f TAGS ID GTAGS GRTAGS GSYMS GPATH tags
rm -rf ./.deps
rm -f Makefile
make[2]: Leaving directory `/home/user/ompi_master/opal/mca/event/external'
Making distclean in mca/event/libevent2022
make[2]: Entering directory `/home/user/ompi_master/opal/mca/event/libevent2022'
Making distclean in libevent
make[3]: Entering directory `/home/user/ompi_master/opal/mca/event/libevent2022/libevent'
make[3]: *** No rule to make target `distclean'.  Stop.
make[3]: Leaving directory `/home/user/ompi_master/opal/mca/event/libevent2022/libevent'
make[2]: *** [distclean-recursive] Error 1
make[2]: Leaving directory `/home/user/ompi_master/opal/mca/event/libevent2022'
make[1]: *** [distclean-recursive] Error 1
make[1]: Leaving directory `/home/user/ompi_master/opal'
make: *** [distclean-recursive] Error 1
```

### CODE REVIEW

**Comment by karasevb** [comment]
@jsquyres @rhc54 @artpol84 please review

**Comment by artpol84** [comment]
:+1:

**Comment by ggouaillardet** [comment]
i made #1687 some time ago, is this PR better ?
note i will rebase #1687 from now since it has some conflicts

**Comment by artpol84** [comment]
I asked @karasevb to fix this coz it was annoying problem to me. 
Didn't knew there is a pr already.
@ggouaillardet please rebase, right now this commit is more compact IMO.

File: opal/mca/pmix/pmix2x/Makefile.am [hunk 1]
```diff
@@ -21,6 +23,12 @@ sources = \
         pmix2x_client.c \
         pmix2x_server_south.c \
         pmix2x_server_north.c
+else
+SUBDISR =
```
Comment (jsquyres) [inline opal/mca/pmix/pmix2x/Makefile.am]: This is a typo.

File: opal/mca/pmix/pmix2x/Makefile.am [hunk 2]
```diff
@@ -10,6 +10,8 @@
 # $HEADER$
 #
 
+if !OPAL_PMIX_USE_EXTERNAL
```
Comment (jsquyres) [inline opal/mca/pmix/pmix2x/Makefile.am]: I have not thought about this deeply, but it strikes me odd to put a `EXTRA_DIST` inside an automake `if` conditional...?

File: opal/mca/pmix/pmix2x/Makefile.am [hunk 3]
```diff
@@ -21,6 +23,12 @@ sources = \
         pmix2x_client.c \
         pmix2x_server_south.c \
         pmix2x_server_north.c
+else
+SUBDISR =
+
+endif
+
+DIST_SUBDIRS = $(SUBDIRS)
```
Comment (jsquyres) [inline opal/mca/pmix/pmix2x/Makefile.am]: `DIST_SUBDIRS` should never be empty; it always needs to contain all the subdirs, and according to the code above, `SUBDIRS` may be empty.  The same is true for the other edits in this PR.

I know you guys didn't know about https://github.com/open-mpi/ompi/pull/1687 before making this PR, but I think I might have said something analogous over there (and then let that PR fall off my radar...)  Ah yes, here it is: https://github.com/open-mpi/ompi/pull/1687#issuecomment-222313382

I'll go have a close look at #1687 now...

File: opal/mca/pmix/pmix2x/Makefile.am (hunk 2 above)
Comment (artpol84) [inline opal/mca/pmix/pmix2x/Makefile.am]: autogen.subdirs is holding "libevent". So I was thinking that it's safer to keep it there.

Comment (jsquyres) [inline opal/mca/pmix/pmix2x/Makefile.am]: But if you put it in the conditional, then if you `make dist`, that file may not be included in the tarball.  That's why the DIST-level stuff *always* has to be filled properly, regardless of which components decide to build/not build.  Make sense?

**Comment by jsquyres** [comment]
@artpol84 @karasevb We just merge #1687 -- can you test to see if that fixes the problem for you?

**Comment by artpol84** [comment]
Ok.

2016-07-06 22:58 GMT+06:00 Jeff Squyres <notifications@github.com>:

> @artpol84 <https://github.com/artpol84> @karasevb
> <https://github.com/karasevb> We just merge #1687
> <https://github.com/open-mpi/ompi/pull/1687> -- can you test to see if
> that fixes the problem for you?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/open-mpi/ompi/pull/1845#issuecomment-230836589>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AHL5Pq7DMcL7OAw-NvcpSgMCxOp1fNj2ks5qS96jgaJpZM4JEEIB>
> .
>



-- 
С Уважением, Поляков Артем Юрьевич
Best regards, Artem Y. Polyakov

**Comment by karasevb** [comment]
I tested and not did see our problem. This problem is fixed in #1687.

**Comment by jsquyres** [comment]
@karasevb Excellent; many thanks.  Thanks, too, for submitting the PR, even though we merged the other one; I appreciate the debugging and coding effort.

**Pull request closed without merging**